### ISSUE TITLE
Vimar connected actuator not correctly recognised

### ISSUE THREAD

**Issue by riccardocossu** [WebThingsIO/zigbee-adapter#239]
I recently installed a bunch of theese actuators; they are recognised as On/Off Switch, and they work great as that, but their functionality is more like a SmartPlug, since they also have a sensor that could be mapped as InstantaneousPowerProperty (not sure about the other properties).
I can fix this myself if given instructions on how to do it, since I can test it at almost every level of detail needed.
I am using latest release of the plugin 0.11.13.
May this be related somehow to #171? I didn't test those devices before, since I only had them recently installed, so I can't really tell whether this broke in latest revision or not.

https://www.vimar.com/en/int/catalog/product/index/code/14593

**Comment by chas-iot** [comment]
please post the ~/.mozilla-iot/data/zigbee-adapter/zb-XXXXXX.json file and indicate which entry relates to an actuator?

**Comment by riccardocossu** [comment]
One of the actuators is f4ce36f44f485b15
[zb.txt](https://github.com/WebThingsIO/zigbee-adapter/files/5583313/zb.txt)

Please note that I haven't been able to tell which property maps to InstantaneousPowerProperty using deconz; I am wondering if that is exposed through zigbee at all (it is exposed through bluetooth, because the app displays it)

**Comment by chas-iot** [comment]
I've annotated the f4ce36f44f485b15 fragment from the file with the translations of the various codes.
TLDR, there's no output cluster exposed for Zigbee that would allow you to set the power level.
Usually this would be 0x008 Level Control but might be one in the range 0x000C to 0x0014
The various GP/Green Power values allow this device to act as a proxy for very low-powered sensors to allow them to send values over Zigbee. Which means not useful or relevant to your question

```javascript
"f4ce36f44f485b15": {
  "addr64": "f4ce36f44f485b15",
  "addr16": "55bb",
  "activeEndpoints": {
    "10": {
      "profileId": "0104",  // ZHA - Common, Home Automation
      "deviceId": "0009",   // Mains Power Outlet
      "deviceVersion": 0,
      "inputClusters": [
        "0000",             // Basic
        "0003",             // Identify
        "0006",             // On/Off
        "0005",             // Scenes
        "0004"              // Groups
      ],
      "outputClusters": [], // I'd expect to see 0008 Level Control here
      "classifierAttributesPopulated": true
    },
    "242": {                // Green Power
      "profileId": "a1e0",  // GP - Green Power
      "deviceId": "0061",   // GP Proxy Minimum
      "deviceVersion": 0,
      "inputClusters": [],
      "outputClusters": [
        "0021"              // Green Power
      ],
      "classifierAttributesPopulated": true
    }
  },
  "properties": {
    "on": {
      "name": "on",
      "value": false,
      "visible": true,
      "title": "On/Off",
      "type": "boolean",
      "@type": "OnOffProperty",
      "profileId": 260,
      "endpoint": 10,
      "clusterId": 6,
      "attr": "onOff",
      "attrId": 0,
      "fireAndForget": false,
      "bindNeeded": false,
      "configReportNeeded": false,
      "initialReadNeeded": false
    }
  },
  "name": "zb-f4ce36f44f485b15-undefined",
  "type": "switch",
  "@type": [
    "OnOffSwitch"
  ],
  "defaultName": "zb-f4ce36f44f485b15-switch",
  "extendedTimeout": true,
  "activeEndpointsPopulated": true,
  "nodeInfoEndpointsPopulated": true,
  "modelId": "Mains_Power_Outlet_v1.0",
  "powerSource": 1,
  "rxOnWhenIdle": 1
},

**Comment by riccardocossu** [comment]
Ok thanks, I will check this out (I think I can upgrade the firmware, it's done through the gateway and the companion app).
In case I manage to do it should I remove the sensor from the UI and pair it again right?

In case there is no upgrade available (or none useful) I will definitely contact vimar since the product is marketed as being both BT and zigbee and the two don't have the same capabilities. I guess most people use the BT version, but still the zigbee one seems incomplete currently

**Comment by riccardocossu** [comment]
The firmware is updated, so I will have to contact Vimar; zigbee is not on par with BT

**Comment by chas-iot** [comment]
after the upgrade, did any new values appear in the Output Clusters?

**Comment by riccardocossu** [comment]
I did not upgrade, since I was already on the latest available version; the configuration I gave yesterday is from the latest fw version. I am waiting for a response by vimar; thanks, you have been very useful to pinpoint the problem.
Where is the list you used to annotate my file? (the one with strandard zigbee endpoints and their explanation)

**Comment by davehylands** [comment]
You can look at the links in this thread: https://github.com/WebThingsIO/zigbee-adapter/issues/251

**Comment by riccardocossu** [comment]
thanks @davehylands, that's exactly what I was looking for.

As for Vimar: they replied like this:
" the view of the instant-consumed power is possible only on the VIEW app, so only with the device in BLE ", saying that this is clearly stated in the instruction manual. I looked it up and it says:
"The VIEW App also makes it possible to view the instant power consumed. The device has
the possibility of being controlled with two different radio standards (exclusive to
one another): Bluetooth mesh (default) or Zigbee (which can be set via the VIEW
Wireless App). The Bluetooth mesh network implies the presence of gateway
20597-19597-16497-14597 while for the dialogue via Zigbee a Zigbee gateway is
required (such as Amazon Echo Plus, Echo Show or Echo Studio)."

Not exactly clear IMHO from those lines that the zigbee mode is crippled...
Btw at this point I guess there is nothing to do, unless an alternative firmware pops up.
Thanks for your help anyway, I will issue a PR on the supported harware page; the smart switch part works flawlessly after all.
